src/map_header_data.c

```diff
@@ -0,0 +1,281 @@
+{
+    return fieldSystem->mapHeaderData->objectEvents;
+}
+
+BOOL MapHeaderData_SetObjectEventPos (FieldSystem * fieldSystem, int localID, u16 x, u16 z)
+{
+    int i;
+    ObjectEvent * objectEvent = fieldSystem->mapHeaderData->objectEvents;
+    u32 numObjectEvents = fieldSystem->mapHeaderData->numObjectEvents;
+
+    for (i = 0; i < numObjectEvents; i++) {
+        if (objectEvent[i].localID == localID) {
+            objectEvent[i].x = x;
+            objectEvent[i].z = z;
+            return TRUE;
+        }
+    }
+
+    GF_ASSERT(FALSE);
+    return FALSE;
+}
+
+BOOL MapHeaderData_SetObjectEventDir (FieldSystem * fieldSystem, int localID, int dir)
+{
+    int i;
+    ObjectEvent * objectEvent = fieldSystem->mapHeaderData->objectEvents;
+    u32 numObjectEvents = fieldSystem->mapHeaderData->numObjectEvents;
+
+    for (i = 0; i < numObjectEvents; i++) {
+        if (objectEvent[i].localID == localID) {
+            objectEvent[i].dir = dir;
+            return TRUE;
+        }
+    }
+
+    GF_ASSERT(FALSE);
+    return FALSE;
+}
+
+BOOL MapHeaderData_SetObjectEventMovementType (FieldSystem * fieldSystem, int localID, int movementType)
+{
+    int i;
+    ObjectEvent * objectEvent = fieldSystem->mapHeaderData->objectEvents;
+    u32 numObjectEvents = fieldSystem->mapHeaderData->numObjectEvents;
+
+    for (i = 0; i < numObjectEvents; i++) {
+        if (objectEvent[i].localID == localID) {
+            objectEvent[i].movementType = movementType;
+            return TRUE;
+        }
+    }
+
+    GF_ASSERT(FALSE);
+    return FALSE;
+}
+
+BOOL MapHeaderData_SetWarpEventPos (FieldSystem * fieldSystem, u16 index, u16 x, u16 z)
+{
+    WarpEvent * warpEvent = fieldSystem->mapHeaderData->warpEvents;
+    warpEvent[index].x = x;
+    warpEvent[index].z = z;
+    return TRUE;
+}
+
+BOOL MapHeaderData_SetWarpEventDestHeaderID (FieldSystem * fieldSystem, u16 index, u16 destHeaderID)
+{
+    WarpEvent * warpEvents = fieldSystem->mapHeaderData->warpEvents;
+    warpEvents[index].destHeaderID = destHeaderID;
+    return TRUE;
+}
+
+BOOL MapHeaderData_SetWarpEventDestWarpID (FieldSystem * fieldSystem, u16 index, u16 destWarpID)
+{
+    WarpEvent * warpEvents = fieldSystem->mapHeaderData->warpEvents;
+    warpEvents[index].destWarpID = destWarpID;
+    return TRUE;
+}
+
+BOOL MapHeaderData_SetBgEventPos (FieldSystem * fieldSystem, u16 index, u16 x, u16 z)
+{
+    BgEvent * bgEvent = MapHeaderData_GetBgEvents(fieldSystem);
+
+    bgEvent += index;
+    bgEvent->x = x;
+    bgEvent->z = z;
+
+    return TRUE;
+}
+
+static void MapHeaderData_ParseEvents (MapHeaderData * data)
+{
+    const u8 * events = (const u8 *)data->tmpEventsBuf;
+
+    // bgEvents
+    data->numBgEvents = *(u32 *)events;
+    events += sizeof(u32);
+    if (data->numBgEvents != 0) {
+        data->bgEvents = (const BgEvent *)events;
+    } else {
+        data->bgEvents = NULL;
+    }
+    events += sizeof(BgEvent) * data->numBgEvents;
```

Comment:
I'm wondering if there is a way to inline this, since it's kind of smelly to have such similar code repeated 4 times with very slight differences. I feel like a macro would be unwieldy.